docs/manual.md

```diff
@@ -813,14 +885,41 @@ In our example, the set `msgs` may contain records of three kinds:
     for some numbers `i` and `j`.
 
 From the perspective of the type checker, the three records shown above have
-three different types. Although we would love to reject this example as an
-ill-typed one, mixing records of different types is a widely-accepted idiom in
-TLA+, for instance, see [Lamport's specification of
+three different types. Mixing records of different types is a widely-accepted
+idiom in TLA+, for instance, see [Lamport's specification of
 Paxos](https://github.com/tlaplus/Examples/blob/master/specifications/Paxos/Paxos.tla).
-Think of records as of C unions, rather than C structs!
+Fortunately, there is a well established precedent for a subtyping relationship
```

Comment:
@konnov or @andrey-kuprianov This is one of the two places where I suggested more substantive elaboration in the manual, I just wanted to call it out to be sure its scrutinized appropriately, since I'm just starting to figure out stuff here and so it's easy for me to misunderstand or make mistakes :)

Reply:
This part I would actually postpone, and leave it as is for now. We can discuss this later, after the 0.7.0 release. My concern is that this brings a bit too much of theory, and might scare off some practice-oriented folks.

Reply:
Cool! This makes sense. If it is useful, perhaps it's better suited  for the document on types. I still think the comparison to C unions is a bit misleading, but perhaps this will make more sense to me after more experience as well :)